internal/zinc-core/src/main/protobuf/zprof.proto

```diff
@@ -0,0 +1,68 @@
+syntax = "proto3";
```

Comment:
It would be nice to add more documentation on the messages and the `zprof.proto` proto file :smiley: 
Especially answering questions like

- why is this message necessary?
- who/what might actually use this message to solve which problem
- a remake to this pull request which marks the initial implementation